common/src/main/java/com/wynntils/core/chat/ChatManager.java

```diff
@@ -0,0 +1,222 @@
+                    newChatLines.push(line);
+                }
+            }
+        } else {
+            // After a NPC dialog screen, Wynncraft sends a "clear screen" with line of ÀÀÀ...
+            // We just ignore that part
+            if (trailingLine.matches("À+")) {
+                newLines.removeFirst();
+                if (newLines.getFirst().equals("§r")) {
+                    newLines.removeFirst();
+                } else {
+                    WynntilsMod.warn("Malformed dialog [#2]: " + newLines.getFirst());
+                }
+            }
+
+            // What remains, if any, are new chat lines
+            newLines.forEach(newChatLines::push);
+        }
+
+        // Register all new chat lines
+        newChatLines.forEach(line -> handleFakeChatLine(line));
+
+        // Update the new dialog
+        handleNpcDialog(dialog);
+    }
+
+    private static void handleFakeChatLine(String codedString) {
+        // This is a normal, single line chat but coded with format codes
+        saveLastChat(codedString);
+        TextComponent message = new TextComponent(codedString);
+        Component updatedMessage = handleChatLine(message, codedString, MessageType.BACKGROUND);
+        // If the message is canceled, we do not need to cancel any packets,
+        // just don't send out the chat message
+        if (updatedMessage == null) return;
+
+        McUtils.sendMessageToClient(updatedMessage);
+    }
+
+    private static void saveLastChat(String codedString) {
+        String msg = ChatFormatting.stripFormatting(codedString);
+        if (!msg.isBlank()) {
+            lastRealChat = msg;
+        }
+    }
+
+    private static RecipientType getRecipientType(Component message, MessageType messageType) {
+        String msg = ComponentUtils.getCoded(message);
+        if (messageType == MessageType.SYSTEM) {
+            // System type messages can only be shouts or "info" messages
+            // We call this MessageType.NORMAL anyway...
+            if (RecipientType.SHOUT.matchPattern(msg, MessageType.NORMAL)) {
+                return RecipientType.SHOUT;
+            }
+            return RecipientType.INFO;
+        } else {
+            for (RecipientType recipientType : RecipientType.values()) {
+                if (recipientType.matchPattern(msg, messageType)) {
+                    return recipientType;
+                }
+            }
+
+            // If no specific recipient matched, it is an "info" message
+            return RecipientType.INFO;
+        }
+    }
+
+    /**
+     * Return a "massaged" version of the message, or null if we should cancel the
+     * message entirely.
+     */
+    private static Component handleChatLine(Component message, String codedMessage, MessageType messageType) {
+        RecipientType recipientType = getRecipientType(message, messageType);
+
+        System.out.println("Handling chat: " + ComponentUtils.getCoded(message) + ", type:" + messageType
```

Comment:
println in non debug feature (or anywhere at all)?

Reply:
Oops. That should not be there.